src/basic/random-util.c

```diff
@@ -93,6 +93,9 @@ int genuine_random_bytes(void *p, size_t n, bool high_quality_required) {
                         }
 
                         already_done = r;
+                } else if (r == 0) {
+                        have_syscall = true;
+                        return -EIO;
```

Comment:
Why not change `r > 0` to `r >= 0` above? It feels unexpected to return `-EIO` here even if `!high_quality_required`.

Reply:
well, this should not really happen in the first place… but handling this like the first if clause would mean that in this case we'd extend with pseudo, which might mean we might return full pseudo randomness as after all we couldn't read anything at all here...

I'd really treat getrandom() returning zero as a complete error, as it shouldn't happen. The only reason I added this patch was that previously returning zero would result in "return -errno" which is really borked, as it returning zero would mean errno is still set to whatever it was set before...

Quite frankly, the getrandom() man page appears to be broken here. It says that getrandom() returns zero in two conditions. one requires GRND_RANDOM to be set (which we never do, so this doesn't apply). the other suggests that this happens when the call is interrupted, but then it also says EINTR is set in this case... Due to the crappy docs i figured we'd rather catch this and treat this as proper error than tape over it...

Reply:
Both caller will either propagate the error or log about it, so if that ever happens, we should know. So this seems OK.